lib/utils/es6_.js

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
+ * the License. A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+const ChangeCase = require('change-case')
+
+const reduce = function(obj, f, accumulator_init) {
+  return Object.keys(obj || {}).reduce((accumulator, key) => {
+    let value = obj[key]
+    return f(accumulator, value, key, obj)
+  }, accumulator_init)
+}
+
+const forEach = function(obj, f) {
+  Object.keys(obj || {}).forEach((key) => {
+    let value = obj[key]
+    f(value, key)
+  });
+}
+
+// Note: This is a crappy version to a certain extent... don't use with Strings, for example...
+const clone = function(object) {
+  return Object.assign(new object.constructor(), object)
+}
+
+const cloneDeep = function(obj) {
```

Comment:
Do we need `clone` and `cloneDeep`? Could we use the `deepExtend` function we have in utils?

Reply:
clone and cloneDeep do slightly different things. deepExtend is slightly different as well - copying onto a specific object.

I think to avoid bugs and refactors it is smart to just have all three used for what they are intended.